warp10/src/main/java/io/warp10/script/ext/token/TOKENDUMP.java

```diff
@@ -131,22 +134,34 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     ReadToken rtoken = null;
     WriteToken wtoken = null;
 
-    byte[] token = OrderPreservingBase64.decode(tokenstr.getBytes(StandardCharsets.UTF_8));
+    if (!customKeys) {
+      try {
+        rtoken = Tokens.extractReadToken(tokenstr);
+      } catch (WarpScriptException wse) {
+        try {
+          wtoken = Tokens.extractWriteToken(tokenstr);
+        } catch (Exception e) {
+          throw new WarpScriptException(getName() + " invalid token.", e);
+        }
+      }
+    } else {
+      byte[] token = OrderPreservingBase64.decode(tokenstr.getBytes(StandardCharsets.UTF_8));
 
-    QuasarTokenDecoder dec = decoder;
+      QuasarTokenDecoder dec = decoder;
 
-    if (null == dec) {
-      long[] lkey = SipHashInline.getKey(SipHashKey);
-      dec = new QuasarTokenDecoder(lkey[0], lkey[1], AESKey);
-    }
+      if (null == dec) {
+        long[] lkey = SipHashInline.getKey(SipHashKey);
+        dec = new QuasarTokenDecoder(lkey[0], lkey[1], AESKey);
+      }
```

Comment:
In case `customKeys` and thus `multikey`, `decoder` is always `null` so you can replace this block with 
```
long[] lkey = SipHashInline.getKey(SipHashKey);
QuasarTokenDecoder dec = new QuasarTokenDecoder(lkey[0], lkey[1], AESKey);
```